public/components/CampaignList/CampaignList.js

```diff
@@ -11,6 +11,40 @@ class CampaignList extends React.Component {
     campaigns: []
   };
 
+  state = {
+    sortOrder: {
+      'name': true, //false = DESC, true = ASC
+      'type': false,
+      'status': false,
+      'actualValue': false,
+      'startDate': false,
+      'endDate': false
+    },
+    headersCssClasses: {
+      'name': 'campaign-list__header-order--desc name',
+      'type': 'campaign-list__header-order type',
+      'status': 'campaign-list__header-order status',
+      'actualValue': 'campaign-list__header-order actualValue',
+      'startDate': 'campaign-list__header-order startDate',
+      'endDate': 'campaign-list__header-order endDate'
+    }
+  };
+
+  setCampaignSort = (c) => {
+    const order = this.state.sortOrder[c] ? 'asc' : 'desc';
+
+    //sort campaign list
+    this.props.uiActions.setCampaignSort(c, this.state.sortOrder[c]);
+    //reset all of the headers' css classes so by default there is always up and down arrow
+    for (let value in this.state.headersCssClasses) {
+      this.state.headersCssClasses[value] = 'campaign-list__header-order ' + value;
```

Comment:
I don't think you need any of this information in the component state... but even if you do, don't modify this.state directly it won't trigger a render, you should call this.setState